--- KeyFileHelper.cpp
+++ KeyFileHelper.new
@@ -8,6 +8,30 @@
 #include <mutex>
 #include <stdlib.h>
 
+#if !defined(GLIB_VERSION_2_40)
+gboolean
+g_key_file_save_to_file (GKeyFile     *key_file,
+                         const gchar  *filename,
+                         GError      **error)
+{
+  gchar *contents;
+  gboolean success;
+  gsize length;
+
+  g_return_val_if_fail (key_file != NULL, FALSE);
+  g_return_val_if_fail (filename != NULL, FALSE);
+  g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
+
+  contents = g_key_file_to_data (key_file, &length, NULL);
+  g_assert (contents != NULL);
+
+  success = g_file_set_contents (filename, contents, length, error);
+  g_free (contents);
+
+  return success;
+}
+#endif
+
 static std::mutex g_key_file_helper_mutex;
 
 KeyFileHelper::KeyFileHelper(const char *filename, bool load)
